app/models/dataset.rb

```diff
@@ -58,5 +62,9 @@ def self.map_keys(buckets)
     buckets.map { |bucket| bucket['key'] }
   end
 
+  def organisation=(organisation)
+    @organisation = Organisation.new(organisation)
```

Comment:
Not sure I follow how you pass in `organisation` to the initialiser here, but on the Organisation "model" it's not expecting an argument...

Reply:
I am including the ActiveModel::Model module into my class so I can initialize the model with a hash of attributes. More info: http://guides.rubyonrails.org/active_model_basics.html#model